llvm/lib/Target/AArch64/AArch64PrologueEpilogue.cpp

```diff
@@ -1308,6 +1307,26 @@ AArch64EpilogueEmitter::AArch64EpilogueEmitter(MachineFunction &MF,
   SEHEpilogueStartI = MBB.end();
 }
 
+void AArch64EpilogueEmitter::moveSPBelowFP(MachineBasicBlock::iterator MBBI,
```

Comment:
This name is confusing, can you name it `deallocateStackSpace` instead and pass in base register `AArch64::FP` as a parameter to the function?

Reply:
I don't think `deallocateStackSpace` is a good name for this function, as that sounds very general, and that's not really what this code does. This function takes a negative offset (which is normally allocating) and moves the SP to the `FP+Offset`. 

I called this `moveSPBelowFP` as that's what this does, it moves the stack pointer to the FP+Offset, while ensuring nothing in the range FP to FP+Offset is (temporarily) deallocated.